filecoin-station/core
# feat: Zinnia Runtime and Peer Checker module

- feat: install Zinnia runtime

- feat: install mod-peer-checker

- feat: run peer checker module via Zinnia

- feat: switch base Docker image to full `node:18`

  >  We cannot use `node:18-alpine` because Zinnia does not support that yet.
  >
  >  I tried to use `node:18-slim`, but Saturn L2 was not able to start in that system.
  >  The Node.js project recommends to use `node:18`, so we should be fine. See https://github.com/nodejs/docker-node#image-variants

- test: move mocha config to config file and add `--exit`
 
- test: make `test/station.js` easier to troubleshoot
 
- test: fix expected Saturn messages

See https://github.com/filecoin-station/zinnia/issues/75

## TODO
- [x] Rebase on top of latest `main`
- [x] Merge metrics from multiple modules
  - Add linux-arm64 builds to Zinnia
    https://github.com/filecoin-station/zinnia/pull/172
  - Rework Dockerfile to use Debian as the base image (Zinnia does not support Alpine yet)
  - Verify that Docker images work, especially on arm64
- [x] https://github.com/filecoin-station/zinnia/issues/148


## Out of Scope

- Access logs of individual Zinnia modules. It's tricky! I'll create a follow-up issue for that. 

- Documentation for Core developers
  - how to update Zinnia Runtime to a new version
  - how to update a Zinnia Module to a new version
  - how to add a new Zinnia Module to Station Core
